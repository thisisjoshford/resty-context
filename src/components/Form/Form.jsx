import React from 'react';
import PropTypes from 'prop-types';
import style from './FormStyle.css';

const Form = ({ url, textBody, onChange, onSubmit }) => (
  <div className={style.Form}>
    <form className={style.Form} onSubmit={onSubmit}>
      
      <label>
        <input 
          type="text"
          name="url"
          className={style.Url}
          placeholder='enter target URL... '
          value={url}
          onChange={onChange}
        >
        </input>
      </label>
      
      <div>
        <label className={style.Radio}>
          <input 
            className={style.RadioButton}
            type="radio" 
            name="method" 
            value="GET" 
            onChange={onChange}
          /> GET 
        </label>
        <label className={style.Radio}>
          <input 
            className={style.RadioButton}
            type="radio" 
            name="method" 
            value="POST"
            onChange={onChange}
          /> POST 
        </label>
        <label className={style.Radio}>
          <input 
            className={style.RadioButton}
            type="radio" 
            name="method" 
            value="PUT"
            onChange={onChange} 
          /> PUT 
        </label>
        <label className={style.Radio}>
          <input 
            className={style.RadioButton}
            type="radio" 
            name="method" 
            value="PATCH" 
            onChange={onChange}
          /> PATCH 
        </label>
        <label className={style.Radio}>
          <input 
            className={style.RadioButton}
            type="radio" 
            name="method" 
            value="DELETE"
            onChange={onChange}
          /> DELETE 
        </label>
      </div>

      <textarea 
        placeholder="JSON Body"
        name="textBody"
        value={textBody}
        onChange={onChange}
      >
      </textarea>

      <button type="submit">GO!</button>

    </form>
  </div>
);

Form.propTypes = {
  url: PropTypes.string.isRequired,
  textBody: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default Form;

