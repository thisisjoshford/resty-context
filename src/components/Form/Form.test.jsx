import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {

  const url = 'http://www.randomapi.com/';

  it('renders Form', () => {
    const wrapper = shallow(
      <Form 
        url={url} 
        textBody=""
        onChange={() => {}} 
        onSubmit={() => {}}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
