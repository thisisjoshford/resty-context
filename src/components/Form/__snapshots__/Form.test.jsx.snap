// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
       URL:
      <input
        name="url"
        onChange={[Function]}
        type="text"
        value="http://www.randomapi.com/"
      />
    </label>
    <div>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="GET"
        />
         GET
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="POST"
        />
         POST
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="PUT"
        />
         PUT
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="PATCH"
        />
         PATCH
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="DELETE"
        />
         DELETE
      </label>
    </div>
    <textarea
      name="textBody"
      onChange={[Function]}
      placeholder="JSON Body"
      value=""
    />
    <button
      type="submit"
    >
      GO!
    </button>
  </form>
</div>
`;
